### PR TITLE
ARROW-15683: [Website] [DataFusion] DataFusion 7.0.0 blog post

### PR DESCRIPTION
**Pull request by matthewmturner**
related: https://github.com/apache/arrow-datafusion/issues/1587

based on crowd sourced content https://docs.google.com/document/d/17uB1GIN58xOehQP5XpJH8J3qA7KDS1WQJzyHzW7L1kU/edit?usp=sharing

### CODE REVIEW

**Comment by matthewmturner** [comment]
FYI @alamb @xudong963

File: _posts/2022-02-14-datafusion-7.0.0.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+# Summary
+
+There have been significant improvements across the board since the 6.0 release which are summarized below.
+
+- DataFusion Crate
+  - The DataFusion crate is in the process of being split into multiple crates in order to decrease compilation times and improve the development experience. To start, datafusion-common (the core DataFusion components) and datafusion-expr (DataFusion expressions, functions, and operators) will be split out.  There will be additional splits after the 7.0 release.
+- Performance Improvements and Optimizations
+  - Arrow’s dyn scalar kernels are now used which enable more efficient operations on DictionaryArrays [#1685](https://github.com/apache/arrow-datafusion/pull/1685)
+  - Switch from std::sync::Mutex to parking_lot::Mutex [#1720](https://github.com/apache/arrow-datafusion/pull/1720)
```
Comment (alamb) [inline _posts/2022-02-14-datafusion-7.0.0.md:100]:
```suggestion
  - Switch from `std::sync::Mutex` to `parking_lot::Mutex` [#1720](https://github.com/apache/arrow-datafusion/pull/1720)
```

File: _posts/2022-02-14-datafusion-7.0.0.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    - New metrics
+      - `Gauge` and `CurrentMemoryUsage` [#1682](https://github.com/apache/arrow-datafusion/pull/1682)
+      - `Spill_count` and `spilled_bytes` [#1641](https://github.com/apache/arrow-datafusion/pull/1641)
+  - New math functions
+    - `Approx_quantile` [#1529](https://github.com/apache/arrow-datafusion/pull/1539)
+    - `stddev` and `variance` (sample and population) [#1525](https://github.com/apache/arrow-datafusion/pull/1525)
+    - `corr` [#1561](https://github.com/apache/arrow-datafusion/pull/1561)
+  - Support decimal type [#1394](https://github.com/apache/arrow-datafusion/pull/1394)[#1407](https://github.com/apache/arrow-datafusion/pull/1407)[#1408](https://github.com/apache/arrow-datafusion/pull/1408)[#1431](https://github.com/apache/arrow-datafusion/pull/1431)[#1483](https://github.com/apache/arrow-datafusion/pull/1483)[#1554](https://github.com/apache/arrow-datafusion/pull/1554)[#1640](https://github.com/apache/arrow-datafusion/pull/1640)
+  - Support for evolved schemas [#1622](https://github.com/apache/arrow-datafusion/pull/1622)[#1709](https://github.com/apache/arrow-datafusion/pull/1709)
```
Comment (alamb) [inline _posts/2022-02-14-datafusion-7.0.0.md:113]:
```suggestion
  - Support for reading parquet files with evolved schemas [#1622](https://github.com/apache/arrow-datafusion/pull/1622)[#1709](https://github.com/apache/arrow-datafusion/pull/1709)
```

File: _posts/2022-02-14-datafusion-7.0.0.md [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+  - Support for registering `DataFrame` as table [#1699](https://github.com/apache/arrow-datafusion/pull/1699)
+  - Suppot `substring` function [#1621](https://github.com/apache/arrow-datafusion/pull/1621)
```
Comment (alamb) [inline _posts/2022-02-14-datafusion-7.0.0.md:115]:
```suggestion
  - Support for the `substring` function [#1621](https://github.com/apache/arrow-datafusion/pull/1621)
```

File: _posts/2022-02-14-datafusion-7.0.0.md [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+  - Support `array_agg(distinct ...)` [#1579](https://github.com/apache/arrow-datafusion/pull/1579)
+  - Support `sort` on unprojected columns [#1415](https://github.com/apache/arrow-datafusion/pull/1415)
+- Additional Integration Points
+  - A new public Expression simplification API [#1717](https://github.com/apache/arrow-datafusion/pull/1717)
+- [DataFusion-Contrib](https://github.com/datafusion-contrib)
+  - A new GitHub organization created as a home for both `DataFusion` extensions and as a testing ground for new features.
```
Comment (alamb) [inline _posts/2022-02-14-datafusion-7.0.0.md:121]: ❤️

File: _posts/2022-02-14-datafusion-7.0.0.md [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+      - [DataFusion-hdsfs-native](https://github.com/datafusion-contrib/datafusion-hdfs-native)
+      - [DataFusion-ObjectStore-s3](https://github.com/datafusion-contrib/datafusion-objectstore-s3)
+    - New Features
+      - [DataFusion-Streams](https://github.com/datafusion-contrib/datafusion-streams)
+- [Arrow2](https://github.com/jorgecarleitao/arrow2)
+  - An [Arrow2 Branch](https://github.com/apache/arrow-datafusion/tree/arrow2) has been created.  There are ongoing discussions in [DataFusion](https://github.com/apache/arrow-datafusion/issues/1532) and [arrow-rs](https://github.com/apache/arrow-rs/issues/1176) about migrating `DataFusion` to `Arrow2`
+
+For the full list of new features with their relevant PRs, see the
+[enhancements section](https://github.com/apache/arrow-datafusion/blob/7.0.0/datafusion/CHANGELOG.md)
+in the changelog.
```
Comment (alamb) [inline _posts/2022-02-14-datafusion-7.0.0.md:134]:
```suggestion
```
I think this is redundant with the lead above

File: _posts/2022-02-14-datafusion-7.0.0.md [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+
+# Documentation and Roadmap
+
+The project's documentation is being consolidated into the official site.  You can find more details there on topics such as the SQL status (TO DO LINK) and a user guide.
```
Comment (alamb) [inline _posts/2022-02-14-datafusion-7.0.0.md:138]:
```suggestion
We are working to consolidate the documentation into the [official site](https://arrow.apache.org/datafusion).  You can find more details there on topics such as the [SQL status](https://arrow.apache.org/datafusion/user-guide/sql/index.html)  and a [user guide](https://arrow.apache.org/datafusion/user-guide/introduction.html#introduction). This is also an area we would love to get help from the broader community.
```

File: _posts/2022-02-14-datafusion-7.0.0.md [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+
+To provide transparency on DataFusion’s priorities to users and developers a three month roadmap will be published at the beginning of each quarter.  This can be found here (TO DO LINK once site is updated).
```
Comment (alamb) [inline _posts/2022-02-14-datafusion-7.0.0.md:140]:
```suggestion
To provide transparency on DataFusion’s priorities to users and developers a three month roadmap will be published at the beginning of each quarter.  This can be found [here[(https://arrow.apache.org/datafusion/specification/roadmap.html).  
```

File: _posts/2022-02-14-datafusion-7.0.0.md [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+in the changelog.
+
+# Documentation and Roadmap
+
+The project's documentation is being consolidated into the official site.  You can find more details there on topics such as the SQL status (TO DO LINK) and a user guide.
+
+To provide transparency on DataFusion’s priorities to users and developers a three month roadmap will be published at the beginning of each quarter.  This can be found here (TO DO LINK once site is updated).  
+
+See full details on DataFusion’s ambitions (TO DO LINK).
+
```
Comment (alamb) [inline _posts/2022-02-14-datafusion-7.0.0.md:143]:
```suggestion
```

File: _posts/2022-02-14-datafusion-7.0.0.md [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+    - [File system access](https://github.com/apache/arrow-datafusion/issues/1702)
+    - [Cluster state](https://github.com/apache/arrow-datafusion/issues/1704)
+- Continued improvements for working with limited resources and large datasets
+  - Memory limited joins[#1599](https://github.com/apache/arrow-datafusion/issues/1599)
+  - Sort-merge join[#141](https://github.com/apache/arrow-datafusion/issues/141)[#1776](https://github.com/apache/arrow-datafusion/pull/1776)
+  - Introduce row based bytes representation [#1708](https://github.com/apache/arrow-datafusion/pull/1708)
+
+# How to Get Involved
+
+If you are interested in contributing to DataFusion, we would love to have you! You
```
Comment (alamb) [inline _posts/2022-02-14-datafusion-7.0.0.md:159]:
```suggestion
If you are interested in contributing to DataFusion, and learning about state of 
the art query processing, we would love to have you join us on the journey! You
```

File: _posts/2022-02-14-datafusion-7.0.0.md [hunk 10]
```diff
@@ -0,0 +1,166 @@
...
+can help by trying out DataFusion on some of your own data and projects and filing bug reports and helping to
+improve the documentation, or contribute to the documentation, tests or code. A list of open issues suitable for
+beginners is [here](https://github.com/apache/arrow-datafusion/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22)
```
Comment (alamb) [inline _posts/2022-02-14-datafusion-7.0.0.md:162]:
```suggestion
can help by trying out DataFusion on some of your own data and projects and let us know how it goes or contribute a PR with documentation, tests or code. A list of open issues suitable for
beginners is [here](https://github.com/apache/arrow-datafusion/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22)
```

**Review by alamb — COMMENTED** [review]
Thank you @matthewmturner  this is looking good -- I'll also send a note to the mailing list / slack channel highlighting this as well

File: _posts/2022-02-14-datafusion-7.0.0.md (hunk 2 above)
Comment (matthewmturner) [reply _posts/2022-02-14-datafusion-7.0.0.md:113]: I had thought #1709 helped generalize schema adaptors to more than just Parquet files.  But i didnt get to dig in too much to that so maybe i misunderstood

File: _posts/2022-02-14-datafusion-7.0.0.md [hunk 11]
```diff
@@ -0,0 +1,154 @@
+---
+layout: post
+title: Apache Arrow DataFusion 6.0.0 Release
```
Comment (xudong963) [inline _posts/2022-02-14-datafusion-7.0.0.md:3]:
```suggestion
title: Apache Arrow DataFusion 7.0.0 Release
```

File: _posts/2022-02-14-datafusion-7.0.0.md [hunk 12]
```diff
@@ -0,0 +1,154 @@
...
+
+# Introduction
+
+[DataFusion](https://arrow.apache.org/datafusion/) is an extensible query execution framework, written in Rust, that uses Apache Arrow as its in-memory format.
+
+When you want to extend your Rust project with [SQL support](https://arrow.apache.org/datafusion/user-guide/sql/sql_status.html), a DataFrame API, or the ability to read and process Parquet, JSON, Avro or CSV data, DataFusion is definitely worth checking out.
+
+DataFusion's  SQL, `DataFrame`, and manual `PlanBuilder` API let users access a sophisticated query optimizer and execution engine capable of fast, resource efficient, and parallel execution that takes optimal advantage of todays multicore hardware. Being written in Rust means DataFusion can offer *both* the safety of dynamic languages as well as the resource efficiency of a compiled language.
+
+The Apache Arrow team is pleased to announce the DataFusion 7.0.0 release. This covers 4 months of development work
```
Comment (xudong963) [inline _posts/2022-02-14-datafusion-7.0.0.md:35]: 🎉

File: _posts/2022-02-14-datafusion-7.0.0.md [hunk 13]
```diff
@@ -0,0 +1,154 @@
...
+     1  Marko Mikulicic
+     1  Max Burke
+     1  Carol (Nichols || Goulding)
+     1  Phillip Cloud
+     1  Rich
+     1  Toby Hede
+     1  Will Jones
+     1  r.4ntix
+     1  rdettai
+```
```
Comment (xudong963) [inline _posts/2022-02-14-datafusion-7.0.0.md:86]: Carlos and xudong.w both are me, lol
```suggestion
    44  Andrew Lamb
    24  Kun Liu
    23  Jiayu Liu
    17  xudong.w
    11  Yijie Shen
     9  Matthew Turner
     7  Liang-Chi Hsieh
     5  Lin Ma
     4  Stephen Carman
     4  James Katz
     4  Dmitry Patsura
     4  QP Hou
     3  dependabot[bot]
     3  Remzi Yang
     3  Yang
     3  ic4y
     3  Daniël Heres
     2  Andy Grove
     2  Raphael Taylor-Davies
     2  Jason Tianyi Wang
     2  Dan Harris
     2  Sergey Melnychuk
     1  Nitish Tiwari
     1  Dom
     1  Eduard Karacharov
     1  Javier Goday
     1  Boaz
     1  Marko Mikulicic
     1  Max Burke
     1  Carol (Nichols || Goulding)
     1  Phillip Cloud
     1  Rich
     1  Toby Hede
     1  Will Jones
     1  r.4ntix
     1  rdettai
```

File: _posts/2022-02-14-datafusion-7.0.0.md [hunk 14]
```diff
@@ -0,0 +1,154 @@
...
+- [DataFusion-Contrib](https://github.com/datafusion-contrib)
+  - A new GitHub organization created as a home for both `DataFusion` extensions and as a testing ground for new features.
+    - Extensions
+      - [DataFusion-Python](https://github.com/datafusion-contrib/datafusion-python)
+      - [DataFusion-Java](https://github.com/datafusion-contrib/datafusion-java)
+      - [DataFusion-hdsfs-native](https://github.com/datafusion-contrib/datafusion-hdfs-native)
+      - [DataFusion-ObjectStore-s3](https://github.com/datafusion-contrib/datafusion-objectstore-s3)
+    - New Features
+      - [DataFusion-Streams](https://github.com/datafusion-contrib/datafusion-streams)
+- [Arrow2](https://github.com/jorgecarleitao/arrow2)
```
Comment (xudong963) [inline _posts/2022-02-14-datafusion-7.0.0.md:125]: 👍

**Review by xudong963 — COMMENTED** [review]
Thanks @matthewmturner

File: _posts/2022-02-14-datafusion-7.0.0.md (hunk 13 above)
Comment (matthewmturner) [reply _posts/2022-02-14-datafusion-7.0.0.md:86]: > Carlos and xudong.w both are me, lol

sry, meant to update that. thx for catching.

File: _posts/2022-02-14-datafusion-7.0.0.md [hunk 15]
```diff
@@ -0,0 +1,154 @@
...
+     1  Carol (Nichols || Goulding)
+     1  Phillip Cloud
+     1  Rich
+     1  Toby Hede
+     1  Will Jones
+     1  r.4ntix
+     1  rdettai
+```
+
+The following section highlights some of the improvements in this release. Of course, many other bug fixes and improvements have also been made and we refer you to the complete [changelog](https://github.com/apache/arrow-datafusion/blob/7.0.0/datafusion/CHANGELOG.md) for the full detail.
```
Comment (liukun4515) [inline _posts/2022-02-14-datafusion-7.0.0.md:88]: The link of the changelog is invalid.

Comment (matthewmturner) [reply _posts/2022-02-14-datafusion-7.0.0.md:88]: I thought it would be valid once officially published - @alamb can you confirm?

Comment (alamb) [reply _posts/2022-02-14-datafusion-7.0.0.md:88]: That is correct -- I will create a `7.0.0` tag in the datafusion-repo once it has been published

Right now, you can preview the log here:
https://github.com/apache/arrow-datafusion/blob/7.0.0-rc2/datafusion/CHANGELOG.md

File: _posts/2022-02-14-datafusion-7.0.0.md (hunk 11 above)
Comment (alamb) [reply _posts/2022-02-14-datafusion-7.0.0.md:3]: that is a *great* catch. 🦅  👁️  👍

File: _posts/2022-02-14-datafusion-7.0.0.md (hunk 2 above)
Comment (alamb) [reply _posts/2022-02-14-datafusion-7.0.0.md:109]: i think @thinkharderdev concluded the schema merging already worked for CSV and Json files, but I may also misunderstand

Comment (thinkharderdev) [reply _posts/2022-02-14-datafusion-7.0.0.md:109]: That's right. Merging already worked for everything except Parquet

File: _posts/2022-02-14-datafusion-7.0.0.md [hunk 16]
```diff
@@ -0,0 +1,153 @@
+---
+layout: post
+title: Apache Arrow DataFusion 7.0.0 Release
+date: "2022-02-14 00:00:00"
```
Comment (alamb) [inline _posts/2022-02-14-datafusion-7.0.0.md:4]:
```suggestion
date: "2022-02-17 00:00:00"
```

I plan to release 7.0.0 in about 1 hour (after the required 72 hour voting time period has elapsed)

Comment (matthewmturner) [reply _posts/2022-02-14-datafusion-7.0.0.md:4]: Updating now

File: _posts/2022-02-14-datafusion-7.0.0.md (hunk 15 above)
Comment (alamb) [reply _posts/2022-02-14-datafusion-7.0.0.md]: https://github.com/apache/arrow-datafusion/releases/tag/7.0.0 is now available, so the change log link now works:

https://github.com/apache/arrow-datafusion/blob/7.0.0/datafusion/CHANGELOG.md

File: _posts/2022-02-17-datafusion-7.0.0.md [hunk 17]
```diff
@@ -0,0 +1,153 @@
...
+
+# Summary
+
+- DataFusion Crate
+  - The DataFusion crate is being split into multiple crates to decrease compilation times and improve the development experience. Initially, `datafusion-common` (the core DataFusion components) and `datafusion-expr` (DataFusion expressions, functions, and operators) have been split out. There will be additional splits after the 7.0 release.
+- Performance Improvements and Optimizations
+  - Arrow’s dyn scalar kernels are now used to enable efficient operations on `DictionaryArray`s [#1685](https://github.com/apache/arrow-datafusion/pull/1685)
+  - Switch from `std::sync::Mutex` to `parking_lot::Mutex` [#1720](https://github.com/apache/arrow-datafusion/pull/1720)
+- New Features
+  - Better support for limiting resource usage
```
Comment (houqp) [inline _posts/2022-02-17-datafusion-7.0.0.md:97]: Perhaps rephrase this as the following since it's a pretty major feature:

```suggestion
  - Support for memory tracking and spilling to disk
```

**Review by houqp — APPROVED** [review]
LGTM, thanks @matthewmturner !

File: _posts/2022-02-17-datafusion-7.0.0.md (hunk 17 above)
Comment (matthewmturner) [reply _posts/2022-02-17-datafusion-7.0.0.md:97]: Ok will do

**Review by liukun4515 — APPROVED** [review]
LGTM
thanks @matthewmturner

**Comment by alamb** [comment]
Shall we publish this one now?

**Comment by matthewmturner** [comment]
Let's do it!

File: _posts/2022-02-17-datafusion-7.0.0.md [hunk 18]
```diff
@@ -0,0 +1,153 @@
+---
+layout: post
+title: Apache Arrow DataFusion 7.0.0 Release
+date: "2022-02-17 00:00:00"
```
Comment (alamb) [inline _posts/2022-02-17-datafusion-7.0.0.md:4]:
```suggestion
date: "2022-02-28 00:00:00"
```

**Comment by alamb** [comment]
🚀

**Comment by kou** [comment]
Could you also update date in path?
("2022-02-17" in https://github.com/apache/arrow-site/blob/master/_posts/2022-02-17-datafusion-7.0.0.md )

**Comment by alamb** [comment]
PR in https://github.com/apache/arrow-site/pull/196
